api/src/server.ts

```diff
@@ -11,25 +13,44 @@ import {
   PORT
 } from './utils/env';
 
+const requestSerializer = (request: FastifyRequest) => ({
+  method: request.method,
+  url: request.url,
+  ip: request.headers['x-forwarded-for']?.toString() || request.ip,
+  hostname: request.hostname,
+  remoteAddress: request.headers['x-forwarded-for']?.toString() || request.ip,
+  remotePort: request.socket.remotePort
+});
+
 const envToLogger = {
   development: {
     transport: {
       target: 'pino-pretty',
       options: {
+        singleLine: true,
         translateTime: 'HH:MM:ss Z',
         ignore: 'pid,hostname'
       }
     },
-    level: FCC_API_LOG_LEVEL || 'info'
+    level: FCC_API_LOG_LEVEL || 'info',
+    serializers: {
+      req: requestSerializer
+    }
   },
   production: {
-    level: FCC_API_LOG_LEVEL || 'info'
-  },
-  test: undefined
+    level: FCC_API_LOG_LEVEL || 'info',
+    serializers: {
+      req: requestSerializer
+    }
+  }
 };
 
 const start = async () => {
-  const fastify = await build({ logger: envToLogger[FREECODECAMP_NODE_ENV] });
+  const fastify = await build({
+    logger: envToLogger[FREECODECAMP_NODE_ENV] ?? true,
+    genReqId: () => randomBytes(8).toString('hex'),
+    disableRequestLogging: true
```

Comment:
With this disabled, the `requestSerializer` never seems to get called. Is the plan to turn this on and use the `requestSerializer`?

Reply:
Actually, the use for `requestSerializer ` will become more evident in later PRs. 

True it is unused as is - but that's because we have been using `fastify.log` everywhere which is fine but not the best way? We should refactor (depending on the context) to use `req.log`, `reply.log` or `req.log.child` etc. which will decorate the log with the "serialized" data automagically.

Reply:
Ah, I checked with fastify.log and never saw it getting called, my bad.